BoltsTests/TaskTests.m

```diff
@@ -605,7 +605,20 @@ - (void)testWaitUntilFinished {
 
     [task waitUntilFinished];
 
+    XCTAssertTrue(task.completed);
     XCTAssertEqualObjects(@"foo", task.result);
+
```

Comment:
Haha. Just noticed this. Yup this is great. 
Maybe split into a separate test and use XCTestExpectation instead of a semaphore here.